packages/read-model/tests/Unit/Product/Listed/ListedProductViewFactoryTest.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Tests\ReadModelBundle\Unit\Product\Listed;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\Pricing\Price;
+use Shopsys\FrameworkBundle\Model\Product\Availability\Availability;
+use Shopsys\FrameworkBundle\Model\Product\Flag\Flag;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Product\ProductCachedAttributesFacade;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Image\ImageViewRepository;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionViewRepository;
+use Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFactory;
+
+class ListedProductViewFactoryTest extends TestCase
+{
+    public function testCreateFromProducts(): void
+    {
+        $productName1 = '22" Sencor SLE 22F46DM4 HELLO KITTY';
+        $productName2 = '32" Philips 32PFL4308';
+        $priceAmount = 10;
+
+        $imageViewRepositoryMock = $this->createImageViewRepositoryMock();
+        $productActionViewRepositoryMock = $this->createProductActionViewRepositoryMock();
+        $domainMock = $this->createDomainMock();
+        $productCachedAttributesFacadeMock = $this->createProductCachedAttributesFacadeMock($priceAmount);
+
+        $listedProductViewFactory = new ListedProductViewFactory($imageViewRepositoryMock, $productActionViewRepositoryMock, $domainMock, $productCachedAttributesFacadeMock);
+
+        $productsMock = [
+            $this->createProductMock(
+                1,
+                $productName1,
+                [
+                    $this->createFlagMock(1),
+                    $this->createFlagMock(5),
+                ]
+            ),
+            $this->createProductMock(
+                2,
+                $productName2,
+                new ArrayCollection([
+                    $this->createFlagMock(4),
+                    $this->createFlagMock(3),
+                ])
+            ),
+        ];
+
+        /** @var \Shopsys\ReadModelBundle\Product\Listed\ListedProductView[] $listedProductViews */
+        $listedProductViews = $listedProductViewFactory->createFromProducts($productsMock);
+
+        $this->assertEquals(1, $listedProductViews[1]->getId());
```

Comment:
This asserting is unfortunately unreadable for me... I have a problem to understand what are we comparing. I hope you can improve it:
```php
$expected = [
1 => new ListedProductView(
 1,
 $productName1,
 'short description,
 ....
),
...
];
$this->assertEqueals($expected, $listedProductViews);
```

What do you think? Will be the new code more readable?

Reply:
I fixed this and I hope its better now :)

Reply:
The data provider is great. Please adjust just the formatting of long method prototype and long object construction to...
```php
public function testCreateFromProduct(
    int $id,
    string $productName,
    ...
)

$expected = new ListedProductView($id,
    $productName,
    $shortDescription,
   ...
);
```